adapter/service.go

```diff
@@ -83,3 +85,7 @@ func (s *service) Specification() *fimptype.Service {
 func (s *service) SendMessage(message *fimpgo.FimpMessage) error {
 	return s.publisher.PublishServiceMessage(s, message)
 }
+
+func (s *service) PublishEvent(event ServiceEvent) {
+	s.publisher.PublishServiceEvent(s, event)
```

Comment:
if in `PublishServiceEvent` we adjust the Event, Address and ServiceName to the `s` anyways, maybe we should take a struct that holds eventType and hasChanged only? Currently it may be quite confusing that we need to create this "big" ServiceEvent struct, pass it to `PublishEvent` that will mutate the majority of the fields anyways.

Reply:
But then it will be impossible to have custom struct on top of embedded pyramid.